run/aem2john.py

```diff
@@ -1,74 +1,118 @@
 #!/usr/bin/env python
 
-# JtR utility to convert native Adobe AEM (Adobe Experience Manager) hashes to
-# an existing JtR hash format.
+"""
+This script is designed to transform Adobe Experience Manager (AEM) hash formats into the hash format used by the John
+the Ripper (JtR) password-cracking tool. It primarily serves as a conversion utility, enabling testing AEM hashes with
+tools that can handle JtR's hash format.
 
-# This software is Copyright (c) 2018, Dhiru Kholia <kholia at kth.se> and it
-# is hereby released to the general public under the following terms:
-#
-# Redistribution and use in source and binary forms, with or without
-# modification, are permitted.
-#
-# See "generateHash" in PasswordUtil.java from the following project,
-# https://github.com/apache/jackrabbit-oak.
+The script reads the original AEM hashes from input files, where each hash is expected to be either in the SHA-256 or
+SHA-512 format and may optionally be prefixed with a username separated by a colon. The script determines its algorithm
+for each hash, extracts the relevant components (salt, iterations, and the hash itself), and reassembles these
+components into the JtR format. 
 
+The JtR hash format is then printed to the standard output. In case of any unrecognized or malformed hashes in the
+input, the script logs a warning and skips to the next hash. 
+
+Usage:
+    python aem2john.py <file1> [<file2> ...]
+
+Each argument should be a file path pointing to a text file containing AEM hashes, one per line.
+
+The script is compatible with both Python 2 and Python 3.
+
+Originally authored by Dhiru Kholia <kholia at kth.se> in 2018 and released to the general public under a permissive
+license allowing free redistribution and modification.
```

Comment:
This wording leaves the current license undefined. Instead of rewording the previous license statement, can you please keep it intact and add your own copyright statement along with Dhiru's? That way, both of you would be copyright holders and the license would remain as it were.

Also, Claudio managed to confuse you about line lengths. Sorry! For comments, let's keep them under 80 (so 79 max). This means you did not need to re-wrap the previously existing lines (Dhiru's) and you need to make yours consistent with those.

(For non-comment code lines, we can let them occasionally go up to 131 where needed.)

Thank you, and sorry for the delay in reviewing your PRs. I intend to proceed quicker now.